### PR TITLE
triumph/presentationnode changes

### PR DESCRIPTION
**Pull request by sundevour**
• fix the nagging lack of indentation on second-to-top level presentation nodes

image

• new triumph display options

image

• marking triumph categories as complete

image

• restyling collections, slightly simpler, with fewer conditional exceptions, and an overall collection count

image

• catalysts: now based on triumphs, though the top level is still filtered traditionally: earned but not completed
styled mostly as records, uses non-generic icons instead of the meaningless generic the triumphs originally had
image

 bhollis 6 hours ago 

The pattern we've been using for Redux props is:

    ProvidedProps for stuff that you set on the component when using it.
    StoreProps for stuff that comes from mapStateToProps.
    DispatchProps for stuff from mapDispatchToProps.

Then at the end there's the type Props = ProvidedProps & StoreProps & DispatchProps. Not only does this make it clearer where everything comes from, it helps to narrow the typechecking for stuff like mapStateToProps so that it's easier to know when things aren't right.

### CODE REVIEW

**Comment by delphiactual** [comment]
With `Hide Completed` off `Top Level` items still will not expand if they are completed see your bottom screenshot `Moments of Triumph`, `Clan`, and `Power Weapons`

File: src/app/collections/PresentationNode.tsx [hunk 1]
```diff
@@ -174,6 +201,22 @@ export default class PresentationNode extends React.Component<Props> {
             </div>
           </div>
         )}
+        {childrenExpanded && presentationNodeHash === 1024788583 && (
+          <div className="presentationNodeOptions">
+            <Checkbox
+              label="Hide Completed"
```
Comment (delphiactual) [inline src/app/collections/PresentationNode.tsx]: Will need to add `i18n`
```suggestion
              label={t('Triumphs.HideCompleted')}
```

File: src/app/collections/PresentationNode.tsx [hunk 2]
```diff
@@ -174,6 +201,22 @@ export default class PresentationNode extends React.Component<Props> {
...
+              name="completedRecordsHidden"
+              value={settings.completedRecordsHidden}
+              onChange={this.onChange}
+            />
+            <Checkbox
+              label="Reveal Redacted"
```
Comment (delphiactual) [inline src/app/collections/PresentationNode.tsx]: same here
```suggestion
              label={t('Triumphs.RevealRedacted')}
```

File: src/app/collections/PresentationNode.tsx [hunk 3]
```diff
@@ -12,7 +12,14 @@ import { deepEqual } from 'fast-equals';
...
 import { scrollToPosition } from 'app/dim-ui/scroll';
 
-interface Props {
+import { setSetting } from '../settings/actions';
+import { RootState } from '../store/reducers';
+import Checkbox from '../settings/Checkbox';
+import { Settings } from '../settings/reducer';
+import { connect } from 'react-redux';
+import { settings } from '../settings/settings';
+
```
Comment (delphiactual) [inline src/app/collections/PresentationNode.tsx]:
```suggestion
import { t } from 'app/i18next-t';

```

Comment (delphiactual) [reply src/app/collections/PresentationNode.tsx]: and add them to `src/locale/dim.json`

Comment (sundevour) [reply src/app/collections/PresentationNode.tsx]: ty!

**Comment by sundevour** [comment]
> With `Hide Completed` - unchecked
> `Top Level` items still will not expand if they are completed see your bottom screenshot `Moments of Triumph`, `Clan`, and `Power Weapons`
over-applied styling made them look un-expandable. should look right now.

**Comment by delphiactual** [comment]
minor UX nit, `Hide Completed` and `Remove Redacted` apply to both `Triumphs` and `Seals` but are hidden under `Triumphs`

**Comment by sundevour** [comment]
mmh that is completely fair. i will explore options but it could get weird to make available in each since the two aren't accordion exclusive

**Comment by delphiactual** [comment]
closes #3306

**Comment by sundevour** [comment]
conflicts resolved, css tuned back up, typescript happy tyvm @bhollis. can you confirm state/store looks good now, and then we are ready for showtime i think

**Comment by sundevour** [comment]
nnnnnevermind TS is still angry, i think my local TS has just given up

**Comment by sundevour** [comment]
we good :100:

**Comment by sundevour** [comment]
@delphiactual can you re-vet for behavior and no surprise weird visual issues?
• i'm formally deferring the "where else should the checkbox display" til a later PR
• completed title bars + hide completed: yes they can slightly expand,
   this is to err towards "hey there's no records" instead of people thinking clicking is broken

**Pull request merged**